assets/js/data/cart/test/notify-quantity-changes.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * External dependencies
+ */
+import { dispatch } from '@wordpress/data';
+import { previewCart } from '@woocommerce/resource-previews';
+import { camelCase, cloneDeep, mapKeys } from 'lodash';
+import { Cart } from '@woocommerce/type-defs/cart';
+import { CartResponse } from '@woocommerce/types';
+
+/**
+ * Internal dependencies
+ */
+import { notifyQuantityChanges } from '../notify-quantity-changes';
+
+jest.mock( '@wordpress/data' );
+
+const mockedCreateInfoNotice = jest.fn();
+dispatch.mockImplementation( ( store ) => {
+	if ( store === 'core/notices' ) {
+		return {
+			createInfoNotice: mockedCreateInfoNotice,
+		};
+	}
+} );
+
+/**
+ * Clones the preview cart and turns it into a `Cart`.
+ */
+const getFreshCarts = (): { oldCart: Cart; newCart: Cart } => {
+	const oldCart = mapKeys(
+		cloneDeep< CartResponse >( previewCart ),
+		( _, key ) => camelCase( key )
+	) as unknown as Cart;
```

Comment:
Why are we using `as` twice with different values here?

Reply:
Without the `as unknown` it complains and says the types don't match.

```
Type 'Dictionary<number | boolean | CartResponseCouponItem[] | CartResponseShippingRate[] | CartResponseShippingAddress | ... 7 more ... | ExtensionsData>' is missing the following properties from type 'Cart': coupons, shippingRates, shippingAddress, billingAddress, and 12 more
```

but we know they do because we know a `CartResponse` with camelCase keys is equivalent to a `Cart`.